google/cloud/storage/examples/storage_quickstart.cc

```diff
@@ -31,14 +31,21 @@ int main(int argc, char* argv[]) {
 
   // Create a client to communicate with Google Cloud Storage. This client
   // uses the default configuration for authentication and project id.
-  gcs::Client client;
-
-  google::cloud::StatusOr<gcs::BucketMetadata> bucket_metadata =
-      client.CreateBucketForProject(
-          bucket_name, project_id,
-          gcs::BucketMetadata()
-              .set_location("us-east1")
-              .set_storage_class(gcs::storage_class::Regional()));
+  google::cloud::StatusOr<gcs::Client> client =
+      gcs::Client::CreateDefaultClient();
+  if (!client) {
+    std::cerr << "Failed to create Storage Client, status=" << client.status()
+              << std::endl;
+    return 1;
+  }
+
+  gcs::BucketMetadata metadata =
+      client
+          ->CreateBucketForProject(
+              bucket_name, project_id,
+              gcs::BucketMetadata()
+                  .set_location("us-east1")
+                  .set_storage_class(gcs::storage_class::Regional()))
```

Comment:
Build is failing here due to these errors:

```
/v/google/cloud/storage/examples/storage_quickstart.cc:42:23: error: no viable conversion from 'StatusOr<google::cloud::storage::v0::BucketMetadata>' to 'gcs::BucketMetadata'
  gcs::BucketMetadata metadata =
                      ^
/v/google/cloud/storage/bucket_metadata.h:490:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'StatusOr<google::cloud::storage::v0::BucketMetadata>' to 'const google::cloud::storage::v0::BucketMetadata &' for 1st argument
class BucketMetadata : private internal::CommonMetadata<BucketMetadata> {
      ^
/v/google/cloud/storage/bucket_metadata.h:490:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'StatusOr<google::cloud::storage::v0::BucketMetadata>' to 'google::cloud::storage::v0::BucketMetadata &&' for 1st argument
/v/google/cloud/storage/examples/storage_quickstart.cc:48:70: error: expected ';' at end of declaration
                  .set_storage_class(gcs::storage_class::Regional()))
                                                                     ^
                                                                     ;
```

Reply:
Whoops. Totally tried to fix this merge conflict and skimmed right over a couple things. Fixed.